src/ln/channelmanager.rs

```diff
@@ -685,10 +693,15 @@ impl ChannelManager {
 						Ok(funding_msg) => {
 							(chan, funding_msg.0, funding_msg.1)
 						},
-						Err(_e) => {
-							//TODO: Push e to pendingevents
+						Err(e) => {
+							mem::drop(channel_state);
+							add_pending_event!(events::Event::HandleOutboundError {
```

Comment:
After previous version, was working on others parts on channelmanager on error handling, and yes it's cumbersome to deal with multiple errors each time, doesn't ease the codebase. So I wrote HandleOutboundError event (really not sure about the name, for me HandleOutboundMsgGenerationError was too specific with DisconnectPeer-only case) which contains a vec of ErrorAction to combine SendErrorMessage, DisconnectPeer and futures errors. That's still hacky, ideally that would be HandleError which hold multiple ErrorAction instead of an Option like right now and having wrapper event like HandleOutboundError avoiding a 1:1 mapping ErrorAction-Events. What do you think about if I refactor first HandleError with a Vec of ErrorAction and after going back to this PR ? Open an issue if you want to think on it before.